src/Simulation/DepletionVoltage.jl

```diff
@@ -0,0 +1,164 @@
+    end
+    
+    inside = findall(p -> p & bulk_bit > 0 && p & update_bit > 0, sim.point_types.data)
+    
+    ϕmin, ϕmax = extrema((ϕρ .+ potential_range[1] * sim.weighting_potentials[contact_id].data)[inside])
+    initial_depletion::Bool = ϕmax - ϕmin < abs(potential_range[1])
+    depletion_voltage::T = NaN
+    start_local_search::T = 0
+    
+    @showprogress for U in potential_range
+        ϕmin, ϕmax = extrema((ϕρ .+ T(U) * sim.weighting_potentials[contact_id].data)[inside])
+        depleted = ϕmax - ϕmin < abs(U)
+        if (initial_depletion && !depleted) || (!initial_depletion && depleted)
+            if T(U)>=0
+                start_local_search = T(U)
+            else
+                start_local_search = T(U)
+            end
+            break
+        end
+    end
+    local_range::AbstractRange = range(start_local_search, last(potential_range), step = step(potential_range))
+    size_data = size(sim.point_types.data)
+    
+    if size_data[2] == 1
+        Ix = CartesianIndex(1,0,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iz,-Iz]
+    else
+        Ix = CartesianIndex(1,0,0)
+        Iy = CartesianIndex(0,1,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iy,-Iy,Iz,-Iz]
+    end
+
+    scale = start_local_search
+    undepleted = 0
+    for (i,idx) in enumerate(inside)
+        if !(0 in [size_data[j] == 1 || idx[j] != 1 && idx[j] != size_data[j] for j in 1:length(idx)])#idx[1]!=1 && idx[1]!=size_data[1] && idx[2]!=1 && idx[2]!=size_data[2] && idx[3]!=1 && idx[3]!=size_data[3]
+            for scale_loc in local_range
+                center_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[idx] + ϕρ[idx]
+                min_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[idx-Ix] + ϕρ[idx-Ix]
+                max_pot = min_pot
+                for neighbour in neighbours
+                    n_idx = idx + neighbour
+                    local_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[n_idx] + ϕρ[n_idx]
+                    if local_pot < min_pot
+                        min_pot = local_pot
+                    elseif local_pot > max_pot
+                        max_pot = local_pot
+                    end
+                end
+                if !(min_pot>center_pot || max_pot<center_pot)
+                    if abs(scale_loc)> abs(scale)
+                        scale = T(scale_loc)
+                        local_range = range(scale, last(potential_range), step = step(potential_range))
+                    end         
+                    break
+                end
+                if scale_loc == local_range[end]
+                    undepleted+=1
+                end         
+            end
+        else
+            deleteat!(inside,i) #Delete indices which are in bulk bit, but still have non defined neighbour (Maybe we can get rid of this, I just kept it for safety)
```

Comment:
Can you comment on this? When did you experience this?

Reply:
Naively, to me it looks like this statements deletes all `i` if one of the axes has length 1.
There might also be a more sophisticated way how to treat `r = 0` in the cylindrical case.

Reply:
No, if the phi axis has length 1, the neighbours are not even checked. This occurs for example in the InvertedCoax, where inside includes indices of the form [1,#,#]. Then a neighbour at [0,#,#] is checked, which throws an error. This is due to the fact that grid points at the boundary of the grid are marked as bulk bits. Therefor those are also inside

Reply:
I don't understand.. If the phi axis would not have length 1, we would still have points inside with [1,#.#] or not?

Reply:
Yes, that is correct. Everything I wrote after the first sentence is completely uncorrelated to the phi axis length.
The issue is, that e.g. [1,#,#] is inside. Because when looking for local min or max, I compare this point to all its neighbours. Meaning also the non-existent neighbour at index [0,#,#]. This throws an error. Therefor I would like [1,#,#] not to be inside

Reply:
And how about just checking all the neighbours except for [0,#,#]?
(So, checking if a neighbour exists and check all the existing neighbours)

Reply:
That is possible too. This would also detect local min or max at the world boundary, so maybe slightly better